.github/src/shared/base.pkl

```diff
@@ -0,0 +1,52 @@
+import "@gha/Workflow.pkl"
+
+import "@gha/actions/Common.pkl"
+import "@gha/actions/Setup.pkl"
+import "@gha/actions/Artifact.pkl"
+
+baseJob: Workflow.Job = new {
+  permissions {
+    checks = "write"
+  }
+  name = "Build and test"
+  `runs-on` = "ubuntu-latest"
+  steps {
+    new Common.Checkout {}
+    new Setup.Java {
+      javaVersion = "17"
+      distribution = "temurin"
+      cache = "gradle"
+    }
+    new {
+      name = "Git fetch origin"
+      // required for spotless job, prevent error: `No such reference 'origin/main'`
+      run = "git fetch origin"
```

Comment:
You probably need a full checkout. The checkout action checks out only the current head. If you need everything (like main) you have to use fetch-depth 0

Reply:
Ah, good to know, thanks!